From 6a3e84f94a93f9ba701920d0237f32397bbf32ad Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 13 Nov 2020 06:09:15 +0200
Subject: [PATCH] [SW-20154] preboot: rework moortec ip thermal code

This patch splits the plaform dependent vs generic
code, such that adding support for gaudi2 becomes
easier.

Change-Id: I3e957e76bee03cc1befcb44ac72dcd373596414e
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2/common.c b/goya2/common.c
index 98ba489..25a07ca 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -25,6 +25,9 @@
 
 #include <hl_boot_if.h>
 #include <goya2/goya2_reg_map.h>
+#include <dcon0_tstd_masks.h>
+#include <dcon0_tstd_regs.h>
+#include "preboot/thermal.h"
 
 #include <preboot.h>
 #include <pll.h>
@@ -536,3 +539,28 @@
 
 	return false;
 }
+
+static struct hl_ts_cfg goya2_hl_ts_data[] = {
+	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+		.use_calib = false, .calib_offset = 10, .hl_tmp_data = { 0 }},
+	{.base = mmDCON0_TSTD_BASE, .alarm_a = 102, .alarm_b =  120,
+		.use_calib = false, .calib_offset = 9, .hl_tmp_data = { 0 }},
+	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }},
+	{.base = mmDCON2_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+		.use_calib = false, .calib_offset = 7, .hl_tmp_data = { 0 }}
+};
+
+void hl_ts_data_init(void)
+{
+	goya2_hl_ts_data[0].freq = hl_pll_get_freq(mmPSOC_SIF_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	goya2_hl_ts_data[1].freq = hl_pll_get_freq(mmPSOC_MME_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	goya2_hl_ts_data[2].freq = hl_pll_get_freq(mmPSOC_SIF_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	goya2_hl_ts_data[3].freq = hl_pll_get_freq(mmPSOC_MME_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	memcpy(hl_ts_data, goya2_hl_ts_data,
+			sizeof(struct hl_ts_cfg) * HL_ASIC_TS_MAX_COUNT);
+}
diff --git a/goya2/tstd.h b/goya2/tstd.h
new file mode 100644
index 0000000..f34c3fe
--- /dev/null
+++ b/goya2/tstd.h
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2020 HabanaLabs Ltd.
+ */
+
+#include <goya2/goya2_reg_map.h>
+#include <goya2_blocks_linux_driver.h>
+#include <dcon0_tstd_masks.h>
+#include <dcon0_tstd_regs.h>
+
+/*TODO: check if the same file can be used across targets of Goya2 */
+#define HL_TS_CMN_SDIF_WDATA_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_SHIFT
+#define HL_TS_CMN_SDIF_WDATA_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_MASK
+#define HL_TS_CMN_SDIF_ADDR_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_SHIFT
+#define HL_TS_CMN_SDIF_ADDR_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_MASK
+#define HL_TS_CMN_SDIF_WRN_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_SHIFT
+#define HL_TS_CMN_SDIF_WRN_MASK		DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_MASK
+#define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
+#define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
+#define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
+	DCON0_TSTD_VM_CMN_SDIF_STATUS_SDIF_BUSY_MASK
+/*
+ * sample_data register width is 16 bits, as per MR74137 spec resolution is
+ * from 8 - 12 bits only.
+ */
+#define HL_TS_00_SDIF_DATA_SMPL_DATA_MASK 0xFFF
+#define HL_TS_CMN_CLK_SYNTH_LO_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_LO_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_MASK
+#define HL_TS_CMN_CLK_SYNTH_HI_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_HI_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_MASK
+#define HL_TS_CMN_CLK_SYNTH_HOLD_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_HOLD_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_MASK
+#define HL_TS_CMN_CLK_SYNTH_ENA_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_ENA_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_MASK
+
+#define HL_TS_CMN_SDIF_STATUS		(mmDCON0_TSTD_TS_CMN_SDIF_STATUS & 0xFF)
+#define HL_TS_CMN_SDIF			(mmDCON0_TSTD_TS_CMN_SDIF & 0xFF)
+#define HL_TS_00_SDIF_DATA		(mmDCON0_TSTD_TS_00_SDIF_DATA & 0xFF)
+#define HL_TS_CMN_CLK_SYNTH		(mmDCON0_TSTD_TS_CMN_CLK_SYNTH & 0xFF)
+#define HL_TS_00_ALARMA_CFG		(mmDCON0_TSTD_TS_00_ALARMA_CFG & 0xFF)
+#define HL_TS_00_ALARMB_CFG		(mmDCON0_TSTD_TS_00_ALARMB_CFG & 0xFF)
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index 69a5644..e32fa31 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -33,6 +33,10 @@
 #include <preboot/fit.h>
 #include "common.h"
 
+#include <dcon0_tstd_masks.h>
+#include <dcon0_tstd_regs.h>
+#include "preboot/thermal.h"
+
 #define FW_WORD_NUM		0x2200	/* FW_SIZE / 2 */
 
 #define mmPCIE_WRAP_VUART_TX_DATA       0xC15F60
@@ -204,3 +208,28 @@
 	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
 	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
 }
+
+static struct hl_ts_cfg goya2_hl_ts_data[] = {
+	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+		.use_calib = false, .calib_offset = 10, .hl_tmp_data = { 0 }},
+	{.base = mmDCON0_TSTD_BASE, .alarm_a = 102, .alarm_b =  120,
+		.use_calib = false, .calib_offset = 9, .hl_tmp_data = { 0 }},
+	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }},
+	{.base = mmDCON2_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
+		.use_calib = false, .calib_offset = 7, .hl_tmp_data = { 0 }}
+};
+
+void hl_ts_data_init(void)
+{
+	goya2_hl_ts_data[0].freq = hl_pll_get_freq(mmPSOC_SIF_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	goya2_hl_ts_data[1].freq = hl_pll_get_freq(mmPSOC_MME_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	goya2_hl_ts_data[2].freq = hl_pll_get_freq(mmPSOC_SIF_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	goya2_hl_ts_data[3].freq = hl_pll_get_freq(mmPSOC_MME_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
+	memcpy(hl_ts_data, goya2_hl_ts_data,
+			sizeof(struct hl_ts_cfg) * HL_ASIC_TS_MAX_COUNT);
+}
diff --git a/goya2_arc_gnu/tstd.h b/goya2_arc_gnu/tstd.h
new file mode 100644
index 0000000..f34c3fe
--- /dev/null
+++ b/goya2_arc_gnu/tstd.h
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2020 HabanaLabs Ltd.
+ */
+
+#include <goya2/goya2_reg_map.h>
+#include <goya2_blocks_linux_driver.h>
+#include <dcon0_tstd_masks.h>
+#include <dcon0_tstd_regs.h>
+
+/*TODO: check if the same file can be used across targets of Goya2 */
+#define HL_TS_CMN_SDIF_WDATA_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_SHIFT
+#define HL_TS_CMN_SDIF_WDATA_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_MASK
+#define HL_TS_CMN_SDIF_ADDR_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_SHIFT
+#define HL_TS_CMN_SDIF_ADDR_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_MASK
+#define HL_TS_CMN_SDIF_WRN_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_SHIFT
+#define HL_TS_CMN_SDIF_WRN_MASK		DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_MASK
+#define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
+#define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
+#define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
+	DCON0_TSTD_VM_CMN_SDIF_STATUS_SDIF_BUSY_MASK
+/*
+ * sample_data register width is 16 bits, as per MR74137 spec resolution is
+ * from 8 - 12 bits only.
+ */
+#define HL_TS_00_SDIF_DATA_SMPL_DATA_MASK 0xFFF
+#define HL_TS_CMN_CLK_SYNTH_LO_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_LO_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_MASK
+#define HL_TS_CMN_CLK_SYNTH_HI_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_HI_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_MASK
+#define HL_TS_CMN_CLK_SYNTH_HOLD_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_HOLD_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_MASK
+#define HL_TS_CMN_CLK_SYNTH_ENA_SHIFT	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_SHIFT
+#define HL_TS_CMN_CLK_SYNTH_ENA_MASK	\
+	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_MASK
+
+#define HL_TS_CMN_SDIF_STATUS		(mmDCON0_TSTD_TS_CMN_SDIF_STATUS & 0xFF)
+#define HL_TS_CMN_SDIF			(mmDCON0_TSTD_TS_CMN_SDIF & 0xFF)
+#define HL_TS_00_SDIF_DATA		(mmDCON0_TSTD_TS_00_SDIF_DATA & 0xFF)
+#define HL_TS_CMN_CLK_SYNTH		(mmDCON0_TSTD_TS_CMN_CLK_SYNTH & 0xFF)
+#define HL_TS_00_ALARMA_CFG		(mmDCON0_TSTD_TS_00_ALARMA_CFG & 0xFF)
+#define HL_TS_00_ALARMB_CFG		(mmDCON0_TSTD_TS_00_ALARMB_CFG & 0xFF)
diff --git a/preboot/thermal_v2.c b/preboot/thermal_v2.c
index 7e85c82..677be85 100644
--- a/preboot/thermal_v2.c
+++ b/preboot/thermal_v2.c
@@ -17,45 +17,7 @@
 
 #include "thermal.h"
 #include "common.h"
-
-#define HL_TS_CMN_SDIF_WDATA_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_SHIFT
-#define HL_TS_CMN_SDIF_WDATA_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_WDATA_MASK
-#define HL_TS_CMN_SDIF_ADDR_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_SHIFT
-#define HL_TS_CMN_SDIF_ADDR_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_ADDR_MASK
-#define HL_TS_CMN_SDIF_WRN_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_SHIFT
-#define HL_TS_CMN_SDIF_WRN_MASK		DCON0_TSTD_TS_CMN_SDIF_SDIF_WRN_MASK
-#define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
-#define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
-#define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
-	DCON0_TSTD_VM_CMN_SDIF_STATUS_SDIF_BUSY_MASK
-/*
- * sample_data register width is 16 bits, as per MR74137 spec resolution is
- * from 8 - 12 bits only.
- */
-#define HL_TS_00_SDIF_DATA_SMPL_DATA_MASK 0xFFF
-#define HL_TS_CMN_CLK_SYNTH_LO_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_LO_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_LO_MASK
-#define HL_TS_CMN_CLK_SYNTH_HI_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_HI_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HI_MASK
-#define HL_TS_CMN_CLK_SYNTH_HOLD_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_HOLD_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYNTH_HOLD_MASK
-#define HL_TS_CMN_CLK_SYNTH_ENA_SHIFT	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_SHIFT
-#define HL_TS_CMN_CLK_SYNTH_ENA_MASK	\
-	DCON0_TSTD_TS_CMN_CLK_SYNTH_CLK_SYTH_EN_MASK
-
-#define HL_TS_CMN_SDIF_STATUS		(mmDCON0_TSTD_TS_CMN_SDIF_STATUS & 0xFF)
-#define HL_TS_CMN_SDIF			(mmDCON0_TSTD_TS_CMN_SDIF & 0xFF)
-#define HL_TS_00_SDIF_DATA		(mmDCON0_TSTD_TS_00_SDIF_DATA & 0xFF)
-#define HL_TS_CMN_CLK_SYNTH		(mmDCON0_TSTD_TS_CMN_CLK_SYNTH & 0xFF)
-#define HL_TS_00_ALARMA_CFG		(mmDCON0_TSTD_TS_00_ALARMA_CFG & 0xFF)
-#define HL_TS_00_ALARMB_CFG		(mmDCON0_TSTD_TS_00_ALARMB_CFG & 0xFF)
+#include <tstd.h>
 
 #define HL_TS_IP_CTRL				0x00
 #define HL_TS_IP_CTRL_AUTO_CONT_MODE		0x108
@@ -85,32 +47,6 @@
 #define HL_PVT_TM_SCRATCH			0x0c
 #define HL_PVT_TM_SCRATCH_PATTERN		0xAAAAAAAA
 #define HL_TS_PVT_INIT_FAILED			0x1
-
-/* Hard coded each of the goya2 on-chip TS device properties */
-static struct hl_ts_cfg goya2_hl_ts_data[] = {
-	{.base = mmDCORE0_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
-		.use_calib = false, .calib_offset = 10, .hl_tmp_data = { 0 }},
-	{.base = mmDCON0_TSTD_BASE, .alarm_a = 102, .alarm_b =  120,
-		.use_calib = false, .calib_offset = 9, .hl_tmp_data = { 0 }},
-	{.base = mmDCORE1_MME_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
-		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }},
-	{.base = mmDCON2_TSTD_BASE, .alarm_a = 102, .alarm_b = 120,
-		.use_calib = false, .calib_offset = 7, .hl_tmp_data = { 0 }}
-};
-
-void hl_ts_data_init(void)
-{
-	goya2_hl_ts_data[0].freq = hl_pll_get_freq(mmPSOC_SIF_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-	goya2_hl_ts_data[1].freq = hl_pll_get_freq(mmPSOC_MME_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-	goya2_hl_ts_data[2].freq = hl_pll_get_freq(mmPSOC_SIF_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-	goya2_hl_ts_data[3].freq = hl_pll_get_freq(mmPSOC_MME_PLL_CTRL_BASE,
-						HL_PLL_DIV3);
-	memcpy(hl_ts_data, goya2_hl_ts_data,
-			sizeof(struct hl_ts_cfg) * HL_ASIC_TS_MAX_COUNT);
-}
 
 void hl_set_calib_coeff(int ts, bool calib_enabled,
 		struct hl_tmp_opmode_info *calib_data)
